crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -1921,6 +1921,20 @@ async fn resolve_relative_request(
 
     let fragment_val = fragment.await?;
 
+    if !options_value.fully_specified && options_value.enable_js_ts_rewriting {
+        // TODO path_pattern might not be a constant?
+        // TODO extension might be empty, or dots in filepath
+        if let Pattern::Constant(s) = path_pattern {
+            if let Some((base, ext)) = s.rsplit_once(".") {
+                if ext == "js" {
+                    new_path = Pattern::Alternatives(vec![Pattern::Constant(base.into()), new_path])
+                }
+            }
+        } else {
+            todo!("enable_js_ts_rewriting");
+        }
```

Comment:
I felt there might be some existing utilities/functions for path handling/etc but I couldn't find anything immediately.